cirq-google/cirq_google/calibration/phased_fsim.py

```diff
@@ -1038,3 +1038,35 @@ def try_convert_sqrt_iswap_to_fsim(gate: cirq.Gate) -> Optional[PhaseCalibratedF
         return PhaseCalibratedFSimGate(cirq.FSimGate(theta=np.pi / 4, phi=0.0), 0.5)
 
     return None
+
+
+def try_convert_gate_to_fsim(gate: cirq.Gate) -> Optional[PhaseCalibratedFSimGate]:
```

Comment:
If you look at my comment below, it would be good to split `prepare_` and `make_` methods to use different converters. However, because of that, it would also be good that whatever is created with `prepare_` stays consistent with `make_` so that you could always reuse the results.

Maybe you could always call the old method first, and do the additional gate resolution only after the result is `None`?

Reply:
Actually, after correction I did today (which ensures theta is in range [0, np.pi)), calling `try_convert_gate_to_fsim` is equivalent to calling `try_convert_sqrt_iswap_to_fsim` and then calling `try_convert_gate_to_fsim` if first call returned `None`.

Reply:
This is a strong requirement for us because if it fails, the characterization and compensation phases are different. Could you add e test specifically to test this condition?

Reply:
After our discussion yesterday this is no longer relevant, since it will be a single function soon.